ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/TransactionServiceOutputsIT.scala

```diff
@@ -56,4 +61,77 @@ class TransactionServiceOutputsIT extends LedgerTestSuite {
       assertEquals("AgreementTextDefault", contract.getAgreementText, "")
     }
   })
+
+  test(
+    "TXVerbosity",
+    "Expose field names only if the verbose flag is set to true",
+    allocate(SingleParty),
+  )(implicit ec => { case Participants(Participant(ledger, party)) =>
+    for {
+      _ <- ledger.create(party, Dummy(party))
+      request = ledger.getTransactionsRequest(Seq(party))
+      verboseTransactions <- ledger.flatTransactions(request.update(_.verbose := true))
+      verboseTransactionTrees <- ledger.transactionTrees(request.update(_.verbose := true))
+      nonVerboseTransactions <- ledger.flatTransactions(request.update(_.verbose := false))
+      nonVerboseTransactionTrees <- ledger.transactionTrees(request.update(_.verbose := false))
+    } yield {
+      assertLabelsAreExposedCorrectly(
+        party,
+        verboseTransactions,
+        verboseTransactionTrees,
+        labelIsNonEmpty = true,
+      )
+      assertLabelsAreExposedCorrectly(
+        party,
+        nonVerboseTransactions,
+        nonVerboseTransactionTrees,
+        labelIsNonEmpty = false,
+      )
+    }
+  })
+
+  private def assertLabelsAreExposedCorrectly(
+      party: Primitive.Party,
+      transactions: Seq[Transaction],
+      transactionTrees: Seq[TransactionTree],
+      labelIsNonEmpty: Boolean,
+  ): Unit = {
+    assert(transactions.nonEmpty, s"$party expected non empty transaction list")
+    assert(transactionTrees.nonEmpty, s"$party expected non empty transaction tree list")
+    val createArgs = flatTransactionFields(transactions)
+    val text = labelIsNonEmpty match {
+      case true => "with"
+      case false => "without"
+    }
+    assert(
+      createArgs.forall(_.label.nonEmpty == labelIsNonEmpty),
+      s"$party expected a contract $text labels, but received $transactions.",
+    )
+    val treeCreateArgs = transactionFields(transactionTreeEvents(transactionTrees))
+    assert(
+      treeCreateArgs.forall(_.label.nonEmpty == labelIsNonEmpty),
+      s"$party expected a contract $text labels, but received $transactionTrees.",
+    )
+  }
+
+  private def transactionTreeEvents(transactionTrees: Seq[TransactionTree]): Seq[CreatedEvent] = {
```

Comment:
Similar comment here, this also makes a lot of assumptions about the transaction tree and will crash if they're not satisfied. I think if you want this to be a general utility you should accept arbitrary trees and filter to the create events within.

Alternatively move it wtihin the test where making assumptions about the transaction is perfectly fine.